app/models/concerns/course/assessment/questions_concern.rb

```diff
@@ -20,4 +20,12 @@ def attempt(submission)
       attempting_answers.fetch(question) { question.attempt(submission) }
     end
   end
+
+  # Returns the questions which do not have a answer or correct answer.
+  #
+  # @param [Course::Assessment::Submission] submission The submission which contains the answers.
+  # @return [Array<Course::Assessment::Question>]
+  def not_correctly_answered(submission)
+    where.not(id: submission.answers.where(correct: true).pluck(:question_id))
```

Comment:
@allenwq can you check if changing the `pluck` to a `select` will cause this to be a subquery?